### PR TITLE
Fix/183 probability dependent metrics do not run not added to pipeline

### PR DESCRIPTION
**Pull request by esherman-credo**
## Change description
1) Add `y_true` to `create_metric_frames()` call in `fairlearn.py` within if-statement for probabilistic metrics
2) Unindented if-block (see line 30 in fix branch), since probabilistic metrics are not necessarily performance metrics (putting `roc_auc_score` in Quickstart.ipynb pipeline alone without other metrics will make it so `roc_auc_score` doesn't get calculated).
3) Added check for whether the specified metric is meant to be for a binary prediction problem (rather than multi-output classification/probabilistic output). Sklearn's `predict_proba()` function outputs an array that has shape `(n_samples, n_classes)`. For binary problems, `y_true` will be `(n_samples, )` in our present handling, and `(n_samples, n_classes)` for a multi-output problem. The former causes a shape mismatch when calling the scoring function and so we need to select the column of `y_prob` corresponding to the positive class.
See discussion of `y_score` here: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html

## Type of change
- [X ] Bug fix (fixes an issue)

## Related issues

> Fix https://github.com/credo-ai/credoai_lens/issues/183

### CODE REVIEW

**Comment by IanAtCredo** [comment]
I don't think we should handle the transformation of `predict_proba` at metric_frame time. We should expect that the predictions created by the model work for the metric. If they don't, I feel the prediction function should be changed. 

Let's table that for more discussion and make this PR just about the bug fix. I'll update.
